src/rust/src/libccxr_exports/gxf.rs

```diff
@@ -0,0 +1,1618 @@
+#[no_mangle]
+unsafe fn copy_ccx_gxf_video_track_from_rust_to_c(
+    ctx: *mut ccx_gxf_video_track,
+    video_track: &CcxGxfVideoTrack,
+) {
+    (*ctx).track_name = string_to_c_array(&video_track.track_name);
+    (*ctx).fs_version = video_track.fs_version;
+    (*ctx).frame_rate = video_track.frame_rate.to_ctype();
+    (*ctx).line_per_frame = video_track.line_per_frame;
+    (*ctx).field_per_frame = video_track.field_per_frame;
+    (*ctx).p_code = video_track.p_code.to_ctype();
+    (*ctx).p_struct = video_track.p_struct.to_ctype();
+}
+#[no_mangle]
+unsafe fn copy_ccx_gxf_ancillary_data_track_from_rust_to_c(
+    ctx: *mut ccx_gxf_ancillary_data_track,
+    ad_track: &CcxGxfAncillaryDataTrack,
+) {
+    (*ctx).track_name = string_to_c_array(&ad_track.track_name);
+    (*ctx).id = ad_track.id;
+    (*ctx).ad_format = ad_track.ad_format.to_ctype();
+    (*ctx).nb_field = ad_track.nb_field;
+    (*ctx).field_size = ad_track.field_size;
+    (*ctx).packet_size = ad_track.packet_size;
+    (*ctx).fs_version = ad_track.fs_version;
+    (*ctx).frame_rate = ad_track.frame_rate;
+    (*ctx).line_per_frame = ad_track.line_per_frame;
+    (*ctx).field_per_frame = ad_track.field_per_frame;
+}
+#[no_mangle]
+unsafe fn copy_ccx_gxf_from_rust_to_c(ctx: *mut ccx_gxf, gxf: &CcxGxf) {
+    (*ctx).nb_streams = gxf.nb_streams;
+    (*ctx).media_name = string_to_c_array(&gxf.media_name);
+    (*ctx).first_field_nb = gxf.first_field_nb;
+    (*ctx).last_field_nb = gxf.last_field_nb;
+    (*ctx).mark_in = gxf.mark_in;
+    (*ctx).mark_out = gxf.mark_out;
+    (*ctx).stream_size = gxf.stream_size;
+
+    (*ctx).cdp_len = gxf.cdp_len;
+    // Handle ad_track
+    if let Some(ref ad_track) = gxf.ad_track {
+        if !(*ctx).ad_track.is_null() {
+            copy_ccx_gxf_ancillary_data_track_from_rust_to_c((*ctx).ad_track, ad_track);
+        }
+    }
+
+    // Handle vid_track
+    if let Some(ref vid_track) = gxf.vid_track {
+        if !(*ctx).vid_track.is_null() {
+            copy_ccx_gxf_video_track_from_rust_to_c((*ctx).vid_track, vid_track);
+        }
+    }
+
+    // Handle cdp buffer
+    if let Some(ref cdp_data) = gxf.cdp {
+        if !(*ctx).cdp.is_null() && !cdp_data.is_empty() {
+            let copy_len = std::cmp::min(cdp_data.len(), gxf.cdp_len);
+            std::ptr::copy_nonoverlapping(cdp_data.as_ptr(), (*ctx).cdp, copy_len);
+        }
+    }
+}
+
+// Helper function to convert C char array to String
+#[no_mangle]
+unsafe fn c_array_to_string(c_array: &[c_char; 256]) -> String {
+    let ptr = c_array.as_ptr();
+    if ptr.is_null() {
+        return String::new();
+    }
+
+    // Find the null terminator or end of array
+    let mut len = 0;
+    while len < 256 && *ptr.add(len) != 0 {
+        len += 1;
+    }
+
+    if len == 0 {
+        return String::new();
+    }
+
+    // Convert to CStr and then to String
+    let slice = std::slice::from_raw_parts(ptr as *const u8, len);
+    String::from_utf8_lossy(slice).into_owned()
```

Comment:
By using this it will convert invalid UTF-8 characters to U+FFFD, whilst C preserves the characters.